File: app-ios/VelaWallet/VelaWallet/Features/Contacts/ContactDetailScreen.swift
//
//  ContactDetailScreen.swift
//  VelaWallet
//
//  Mobile contact detail (spec 018 FR-002, mocks C2 / C2s): back chevron +
//  edit pencil, identicon hero, name, short address, group chips, the three
//  action cards, the full mono address block, 最近往来 over spec-015
//  ActivityRow instances, and the centered destructive 删除联系人 which
//  raises the confirm sheet (C2s).
//
//  Reused from spec 015: IdenticonAvatar, ActionButtonRow, WalletSectionHeader,
//  ActivityRowView, WalletEmptyState (the no-activity edge case).
//

import SwiftUI

struct ContactDetailScreen: View {
    @Environment(\.theme) private var theme

    let model: ContactDetailModel
    var onBack: () -> Void = {}

    @State private var sheetShown = false

    var body: some View {
        VStack(spacing: Tokens.Space.s0) {
            navBar
            ScrollView {
                VStack(alignment: .leading, spacing: Tokens.Space.s0) {
                    hero
                    ActionButtonRow(items: [
                        ActionCardItem(icon: .arrowUpRight, label: model.actions.send),
                        ActionCardItem(icon: .arrowDownLeft, label: model.actions.receive),
                        ActionCardItem(icon: .qrCode, label: model.actions.qr),
                    ])
                    .padding(.top, Tokens.Space.s24)

                    hairline.padding(.top, Tokens.Space.s24)

                    AddressBlock(
                        label: model.addressLabel,
                        lines: model.addressLines,
                        copyLabel: model.copyLabel,
                        copiedLabel: model.copiedLabel
                    )
                    .padding(.top, Tokens.Space.s24)

                    hairline.padding(.top, Tokens.Space.s24)

                    WalletSectionHeader(title: model.activityTitle, action: model.activityAction)
                        .padding(.top, Tokens.Space.s24)
                    activity

                    deleteAction
                        .padding(.top, Tokens.Space.s32)
                }
                .padding(.horizontal, Tokens.Layout.screenPaddingX)
                .padding(.bottom, Tokens.Space.s32)
            }
        }
        .background(theme.bgBase.ignoresSafeArea())
        .environment(\.walletTextScale, model.textScale)
        .sheet(isPresented: $sheetShown) {
            if let sheet = model.sheet {
                ActionMenuSheet(model: sheet, onItem: { _ in }, onCancel: { sheetShown = false })
                    .environment(\.walletTextScale, model.textScale)
            }
        }
        .onAppear { sheetShown = model.sheet != nil }
    }

    // MARK: - Chrome

    private var navBar: some View {
        HStack(spacing: Tokens.Space.s12) {
            Button(action: onBack) {
                LucideIcon(.chevronLeft, size: LucideIconSize.action)
                    .foregroundStyle(theme.fgBase)
                    .frame(width: Tokens.Layout.hitTarget, height: Tokens.Layout.hitTarget)
                    .contentShape(Rectangle())
            }
            .buttonStyle(.plain)
            .accessibilityLabel(model.backLabel)
            Spacer(minLength: Tokens.Space.s12)
            Button {} label: {
                LucideIcon(.pencil, size: LucideIconSize.menuRow)
                    .foregroundStyle(theme.fgBase)
                    .frame(width: Tokens.Layout.hitTarget, height: Tokens.Layout.hitTarget)
                    .contentShape(Rectangle())
            }
            .buttonStyle(.plain)
            .accessibilityLabel(model.editLabel)
        }
        .padding(.horizontal, Tokens.Space.s12)
    }

    private var hero: some View {
        VStack(spacing: Tokens.Space.s0) {
            IdenticonAvatar(seed: model.contact.addressFull, size: ContactsGeometry.detailAvatar)
            Text(verbatim: model.contact.name)
                .typeRole(Typography.title.scaled(model.textScale))
                .foregroundStyle(theme.fgBase)
                .multilineTextAlignment(.center)
                .padding(.top, Tokens.Space.s16)
            Text(verbatim: model.contact.addressDisplay)
                .monoRole(Typography.monoAddressDetail.scaled(model.textScale))
                .foregroundStyle(theme.fgMuted)
                .padding(.top, Tokens.Space.s8)
            GroupChips(chips: model.chips, addLabel: model.addChip)
                .padding(.top, Tokens.Space.s12)
        }
        .frame(maxWidth: .infinity)
        .padding(.top, Tokens.Space.s16)
    }

    @ViewBuilder private var activity: some View {
        if model.activity.isEmpty, let empty = model.activityEmpty {
            WalletEmptyState(icon: .inbox, model: empty)
        } else {
            VStack(spacing: Tokens.Space.s0) {
                ForEach(Array(model.activity.enumerated()), id: \.element.id) { index, row in
                    if index > 0 {
                        Rectangle()
                            .fill(theme.borderBase)
                            .frame(height: Tokens.BorderWidth.hairline)
                            .padding(.leading, WalletGeometry.rowDividerInset)
                    }
                    ActivityRowView(model: row)
                }
            }
        }
    }

    private var deleteAction: some View {
        Button {
            sheetShown = true
        } label: {
            Text(verbatim: model.deleteLabel)
                .typeRole(Typography.button.scaled(model.textScale))
                .foregroundStyle(theme.errorBase)
                .frame(maxWidth: .infinity)
                .frame(minHeight: Tokens.Control.md)
                .contentShape(Rectangle())
        }
        .buttonStyle(.plain)
    }

    private var hairline: some View {
        Rectangle()
            .fill(theme.borderBase)
            .frame(height: Tokens.BorderWidth.hairline)
    }
}

#Preview("C2 detail dark") {
    ContactDetailScreen(model: ContactsFixtures.buildMobileState(.c2, loc: ContactsPreviewData.loc).detail!)
        .themed(.dark)
        .environment(\.identiconProvider, .previewSafe)
        .environment(\.lucideIconProvider, .previewSafe)
}

#Preview("C2 detail light") {
    ContactDetailScreen(model: ContactsFixtures.buildMobileState(.c2, loc: ContactsPreviewData.loc).detail!)
        .themed(.light)
        .environment(\.identiconProvider, .previewSafe)
        .environment(\.lucideIconProvider, .previewSafe)
}

File: app-android/vela-wallet/app/src/main/java/app/getvela/wallet/feature/contacts/ContactDetailScreen.kt
package app.getvela.wallet.feature.contacts

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.navigationBarsPadding
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.statusBarsPadding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import app.getvela.wallet.core.designsystem.components.VelaIcons
import app.getvela.wallet.core.designsystem.theme.VelaTheme
import app.getvela.wallet.core.designsystem.tokens.VelaFontFamily
import app.getvela.wallet.core.designsystem.tokens.VelaFontWeight
import app.getvela.wallet.core.designsystem.tokens.VelaMonoFontFamily
import app.getvela.wallet.core.designsystem.tokens.VelaSizing
import app.getvela.wallet.core.designsystem.tokens.VelaSpacing
import app.getvela.wallet.core.designsystem.tokens.VelaTextSize
import app.getvela.wallet.feature.contacts.components.AddressBlock
import app.getvela.wallet.feature.contacts.components.ContactsMetrics
import app.getvela.wallet.feature.contacts.components.ContactsNavHeader
import app.getvela.wallet.feature.contacts.components.DeleteConfirmSheet
import app.getvela.wallet.feature.contacts.components.DestructiveTextButton
import app.getvela.wallet.feature.contacts.components.EmptyStateCta
import app.getvela.wallet.feature.contacts.components.GroupChips
import app.getvela.wallet.feature.contacts.components.Hairline
import app.getvela.wallet.feature.contacts.components.glyphFor
import app.getvela.wallet.feature.wallet.components.ActionButtonItem
import app.getvela.wallet.feature.wallet.components.ActionButtonRow
import app.getvela.wallet.feature.wallet.components.ActivityRow
import app.getvela.wallet.feature.wallet.components.IdenticonAvatar
import app.getvela.wallet.feature.wallet.components.SectionHeader

/**
 * Contact detail (mocks C2 / C2s): back + pencil header, hero identicon, name,
 * short address, group chips, the reused three-card action dock, the mono
 * 地址 block, 最近往来 over spec-015 ActivityRows, and the centred destructive
 * 删除联系人 that raises the confirmation sheet.
 */
@Composable
fun ContactDetailScreen(
    model: ContactsHomeModel,
    modifier: Modifier = Modifier,
    actions: ContactsActions = ContactsActions(),
) {
    val detail = model.detail ?: return
    val colors = VelaTheme.colors
    Box(
        modifier = modifier
            .fillMaxSize()
            .background(colors.bgBase),
    ) {
        Column(
            modifier = Modifier
                .fillMaxSize()
                .statusBarsPadding()
                .verticalScroll(rememberScrollState())
                .navigationBarsPadding()
                .padding(horizontal = VelaSizing.screenPaddingX),
        ) {
            Spacer(modifier = Modifier.height(VelaSpacing.md))
            ContactsNavHeader(
                backContentDescription = model.backLabel,
                trailingIcon = VelaIcons.Pencil,
                trailingContentDescription = detail.editLabel,
                onBack = { actions.onAction("contacts.back") },
                onTrailing = { actions.onAction("contacts.edit") },
            )

            Spacer(modifier = Modifier.height(VelaSpacing.xl3))
            Column(
                modifier = Modifier.align(Alignment.CenterHorizontally),
                horizontalAlignment = Alignment.CenterHorizontally,
            ) {
                IdenticonAvatar(
                    seed = detail.contact.addressFull,
                    size = ContactsMetrics.heroAvatar,
                    contentDescription = detail.contact.name,
                )
                Spacer(modifier = Modifier.height(VelaSpacing.xl))
                Text(
                    text = detail.contact.name,
                    color = colors.fgBase,
                    fontFamily = VelaFontFamily,
                    fontWeight = VelaFontWeight.bold,
                    fontSize = VelaTextSize.xl3,
                    maxLines = 1,
                )
                Spacer(modifier = Modifier.height(VelaSpacing.md))
                Text(
                    text = detail.contact.addressDisplay,
                    color = colors.fgMuted,
                    fontFamily = VelaMonoFontFamily,
                    fontWeight = VelaFontWeight.regular,
                    fontSize = VelaTextSize.base,
                    maxLines = 1,
                )
                Spacer(modifier = Modifier.height(VelaSpacing.lg))
                GroupChips(
                    model = detail.chips,
                    onGroup = { actions.onAction("contacts.sectionGroups") },
                    onAdd = { actions.onAction("contacts.moveGroup") },
                )
            }

            Spacer(modifier = Modifier.height(VelaSpacing.xl3))
            ActionButtonRow(
                items = detail.actions.map { action ->
                    ActionButtonItem(
                        icon = glyphFor(action.icon),
                        label = action.label,
                        onClick = { actions.onAction("contacts.action." + action.icon.name) },
                    )
                },
            )

            Spacer(modifier = Modifier.height(VelaSpacing.xl3))
            Hairline()
            Spacer(modifier = Modifier.height(VelaSpacing.xl3))
            AddressBlock(
                model = detail.address,
                onCopy = { actions.onAction("contacts.copyAddress") },
            )

            Spacer(modifier = Modifier.height(VelaSpacing.xl3))
            SectionHeader(
                title = detail.activity.title,
                action = detail.activity.action,
                onAction = { actions.onAction("history.filterAll") },
            )
            if (detail.activity.rows.isEmpty()) {
                detail.activity.empty?.let {
                    EmptyStateCta(model = it, icon = VelaIcons.Inbox)
                }
            } else {
                detail.activity.rows.forEach { row -> ActivityRow(model = row) }
            }

            Spacer(modifier = Modifier.height(VelaSpacing.xl4))
            DestructiveTextButton(
                label = detail.deleteLabel,
                onClick = { actions.onAction("contacts.deleteContact") },
            )
            Spacer(modifier = Modifier.height(VelaSpacing.xl4))
        }
    }

    model.deleteConfirm?.let { confirm ->
        DeleteConfirmSheet(
            model = confirm,
            onDismiss = actions.onDismissMenu,
            onConfirm = { actions.onAction("contacts.delete") },
        )
    }
}
